packages/contracts-bedrock/src/L2/GasPriceOracle.sol

```diff
@@ -53,6 +53,26 @@ contract GasPriceOracle is ISemver {
         return _getL1FeeBedrock(_data);
     }
 
+    /// @notice returns an upper bound for the L1 fee for a given transaction size.
+    /// It is provided for callers who wish to estimate L1 transaction costs in the
+    /// write path, and is much more gas efficient than `getL1Fee`.
+    /// It assumes the worst case of fastlz upper-bound which covers %99.99 txs.
+    /// @param _unsignedTxSize Unsigned fully RLP-encoded transaction size to get the L1 fee for.
+    /// @return L1 estimated upper-bound fee that should be paid for the tx
+    function getL1FeeUpperBound(uint256 _unsignedTxSize) external view returns (uint256) {
+        require(isFjord, "GasPriceOracle: getL1FeeUpperBound only supports Fjord");
```

Comment:
i think we probably don't need this given it won't be available before fjord, but don't feel strongly